lib/Zonemaster/Backend/DB/PostgreSQL.pm

```diff
@@ -225,11 +225,13 @@ sub get_test_history {
             undelegated,
             creation_time at time zone current_setting('TIMEZONE') at time zone 'UTC' as creation_time
         FROM test_results
-        WHERE domain=" . $dbh->quote( $p->{frontend_params}->{domain} ) . " $undelegated
-        ORDER BY id DESC
-        OFFSET $p->{offset} LIMIT $p->{limit}";
+        WHERE progress = 100 AND domain = ? ] . $undelegated . q[
```

Comment:
The `IS NOT DISTINCT FROM` operator could be used here to avoid concatenation of SQL syntax.